Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -869,6 +869,42 @@ namespace AssetProcessor
         return m_lfsPointerFileValidator->IsLfsPointerFile(filePath);
     }
 
+    AZ::Outcome<QString, void> AssetProcessorManager::GetFirstTopmostSourceAssetForProductPath(const AZStd::string& productPath)
```

Comment:
I think this is providing the same functionality as AssetUtilities::GetTopLevelSourceForIntermediateAsset

Reply:
I would also ensure we define 'topmost' in the comment. FirstTopMost is a confusing mash of naming to me and I feel either needs a better name or a comment explaining the why of this method

Reply:
I've switched to using GetTopLevelSourceForIntermediateAsset. This also means I've removed the phrase "topmost" so we're now consistently using TopLevel instead.